### ISSUE TITLE
Implement Intent Classification and Instruction Routing

### ISSUE THREAD

**Issue by marctjones** [marctjones/klareco#693]
## Goal
Enable Klareco to follow instructions by classifying user intent from AST structure and routing to appropriate expert modules.

## Motivation
To handle diverse tasks (QA, summarization, translation, generation), Klareco needs to understand **what the user wants** and dispatch to the right processing pipeline.

### Intent Classifier (~5M params)
```python
class IntentClassifier:
    """Classify intent from instruction AST"""
    
    # Deterministic patterns (no learning needed for obvious cases)
    INTENT_PATTERNS = {
        'extractive_qa': lambda ast: ast['verbo']['radiko'] in ['respond', 'kiu', 'kio'],
        'summarization': lambda ast: ast['verbo']['radiko'] in ['resumig', 'klarig'],
        'translation': lambda ast: ast['verbo']['radiko'] == 'traduk',
        'generation': lambda ast: ast['verbo']['radiko'] in ['kre', 'skrib', 'komplet'],
    }
    
    # Learned classifier for ambiguous cases
    def classify(self, ast):
        # Try deterministic patterns first
        for intent, pattern in self.INTENT_PATTERNS.items():
            if pattern(ast):
                return intent
        
        # Fall back to learned classifier
        ast_embedding = self.tree_encoder(ast)
        intent_logits = self.classifier(ast_embedding)
        return self.intent_vocab[intent_logits.argmax()]
```

### Instruction Handler
```python
class InstructionHandler:
    """Parse instructions and route to appropriate AST operations"""
    
    def __init__(self):
        self.parser = Parser()
        self.intent_classifier = IntentClassifier()
        self.experts = {
            'extractive_qa': ExtractiveMachine(),      # Already exists
            'summarization': Summarizer(),              # Already exists
            'translation': Translator(),                # TODO
            'generation': ASTGenerator(),               # Issue #692
        }
    
    def handle(self, instruction: str) -> str:
        # Parse instruction to AST
        instruction_ast = self.parser.parse(instruction)
        
        # Classify intent from AST structure
        intent = self.intent_classifier.classify(instruction_ast)
        
        # Route to expert
        response_ast = self.experts[intent].execute(instruction_ast)
        
        # Deparse to text
        return deparse(response_ast)
```

## Training Data
Create instruction-response pairs in Esperanto:
```jsonl
{"instruction": "Respondu: Kiu fondis Esperanton?", "intent": "extractive_qa", "response": "Zamenhof fondis Esperanton."}
{"instruction": "Resumigu la tekston pri Zamenhof.", "intent": "summarization", "response": "..."}
{"instruction": "Traduku al la angla: Mi amas Esperanton.", "intent": "translation", "response": "I love Esperanto."}
{"instruction": "Kompletigu: Mi amas la hun...", "intent": "generation", "response": "hundon."}
```

## Implementation Plan
1. **Phase 1: Deterministic patterns** (0 params)
   - Implement pattern matching for common verbs
   - Test on basic instructions
   
2. **Phase 2: Learned classifier** (~5M params)
   - Collect instruction-intent pairs
   - Train TreeLSTM-based classifier
   - Handle ambiguous cases

3. **Phase 3: Integration**
   - Extend `orchestrator.py` with InstructionHandler
   - Add CLI: `klareco instruct "Respondu je..."`
   - Add tests for intent classification accuracy

## Success Criteria
- [ ] Intent classifier implemented (deterministic + learned)
- [ ] Intent classification accuracy >95% on test set
- [ ] InstructionHandler routes to correct expert modules
- [ ] Training script: `scripts/train_intent_classifier.sh`
- [ ] CLI integration: `klareco instruct "<instruction>"`
- [ ] Documentation: Supported instruction types

## Dependencies
- Parser (already exists)
- TreeLSTMEncoder (already exists)
- ExtractiveMachine (#680, #690)
- Summarizer (already exists)
- ASTGenerator (#692)

## Related Issues
- #680: Extractive answer generation
- #690: Abstractive answer generation
- #692: AST generation
- Currently: `orchestrator.py` does basic intent routing, extend it

## Training Pipeline Position
```
Instruction Text → Parser → AST → [Intent Classifier] → Expert Module → Response
```

## Supported Instructions (Initial)
- **Extractive QA**: "Respondu je...", "Kiu...", "Kio...", "Kiam...", "Kie..."
- **Summarization**: "Resumigu...", "Klarigu..."
- **Translation**: "Traduku al..."
- **Generation**: "Kompletigu...", "Skribu...", "Kreu..."

## Future Extensions
- Multi-intent instructions (e.g., "Resumigu kaj traduku...")
- Clarification questions when intent unclear
- Confidence thresholds for routing

**Comment by marctjones** [comment]
Closing as part of a 2026-09-08 backlog cleanup (see PLAN.md, DESIGN.md, CLAUDE.md's merge gate).

This issue predates the merge gate (adopted 2026-07-13) and the AST-native orchestrator rewrite. It describes work against the pre-orchestrator RAG pipeline or the learned model stack (M1/M2 embeddings, compositional embeddings, entity classifiers, neural rerankers) that was pruned from the repo -- klareco/embeddings/, klareco/models/, and klareco/summarization/ no longer exist. It has no milestone and was never benchmarked, matching exactly the failure mode CLAUDE.md documents (25 capability issues filed 2026-05-26, 8 implemented, none benchmarked).

Labeled archived-2026-09 so this batch is recoverable: gh issue list --state closed --label archived-2026-09. If this capability is still wanted, re-file it against the current architecture naming the benchmark metric and test the merge gate requires.
